### PR TITLE
Display user-friendly messages for JSON RPC errors

### PR DESCRIPTION
**Pull request by denbite**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/near/near-api-js/blob/master/CONTRIBUTING.md).
- [x] Commit messages follow the [conventional commits](https://www.conventionalcommits.org/) spec
- [x] **If this is a code change**: I have written unit tests.
- [x] **If this changes code in a published package**: I have run `pnpm changeset` to create a `changeset` JSON document appropriate for this change.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Added error parsing to `JsonRpcProvider` when the error message is under the `response.result` key to prevent users from seeing raw errors

#### Before
image
image
image

#### After

image
image
image

## Related issues/PRs

https://github.com/near/near-api-js/pull/1028

### CODE REVIEW

**Review by gtsonevv — DISMISSED** [review]

File: .changeset/red-mice-argue.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+"@near-js/providers": patch
+"@near-js/utils": patch
+---
+
+Display user-friendly messages for JSON RPC errors:
+
+- MethodNotFound
+- CodeDoesNotExist
+- AccessKeyDoesNotExist
+- AccountDoesNotExist
```
Comment (vikinatora) [inline .changeset/red-mice-argue.md:11]: I'd say that the changes should be minor instead of patch

**Review by vikinatora — DISMISSED** [review]
lgtm, just adjust the changeset

**Comment by denbite** [comment]
> lgtm, just adjust the changeset

done

**Review by vikinatora — APPROVED** [review]

**Comment by vikinatora** [comment]
Hey @denbite, please resolve the CI issues and we're ready to merge!

**Review by vikinatora — DISMISSED** [review]

**Comment by vikinatora** [comment]
@denbite there are still two test errors
image

**Comment by vikinatora** [comment]
@denbite there are errors again, you can run `pnpm run test` from the root directory in order to run the tests locally. This will make sure you're not revisiting the PR every day

**Comment by denbite** [comment]
let's run checks again, this time it will pass

**Comment by denbite** [comment]
I had difficulty running tests locally on the M1 chip, but I was able to fix it and opened the Pull Request to merge it
let's make sure it will be merged asap #1283

**Review by vikinatora — APPROVED** [review]

File: packages/utils/fetch_error_schema.js [hunk 2]
```diff
@@ -2,8 +2,8 @@ const https = require('https');
 const fs = require('fs');
 
 const ERROR_SCHEMA_URL =
-    'https://raw.githubusercontent.com/near/nearcore/master/chain/jsonrpc/res/rpc_errors_schema.json';
-const TARGET_SCHEMA_FILE_PATH = `${process.argv[2] ||  process.cwd()}/src/errors/rpc_error_schema.json`;
+    'https://raw.githubusercontent.com/nearprotocol/nearcore/4c1149974ccf899dbcb2253a3e27cbab86dc47be/chain/jsonrpc/res/rpc_errors_schema.json';
+const TARGET_SCHEMA_FILE_PATH = `${process.argv[2] ||  process.cwd()}/src/errors/rpc_error_schema.json'`;
```
Comment (vikinatora) [inline packages/utils/fetch_error_schema.js:6]: Not sure if we should use a rpc schema from 2020

Comment (denbite) [reply packages/utils/fetch_error_schema.js]: the newest version was causing a few tests to crush, but I managed to fix them

**Comment by denbite** [comment]
I believe it should work now, have tested all packages

**Comment by denbite** [comment]
@vikinatora do you know if this RPC https://rpc.ci-testnet.near.org is used for running tests here?
I had to remove a part of the code in one of the tests to pass it locally (this seemed like a mandatory update after upgrading the RPC schema), but here it hadn't passed anyway

image

so I reverted changes back in  https://github.com/near/near-api-js/pull/1275/commits/2815f1ea5630f5365c87799cb06868f2015a0b7c

**Review by vikinatora — APPROVED** [review]

**Pull request merged**